### PR TITLE
OSDOCS-1192 Configuring registry storage for Azure Government

### PR DESCRIPTION
**Pull request by bmcelvee**
https://issues.redhat.com/browse/OSDOCS-1192

### CODE REVIEW

**Comment by bmcelvee** [comment]
@dmage would you mind reviewing this PR, please? It's adding the registry storage configuration for Azure Government Cloud. I'm not sure exactly how it differs from the user-provisioned infrastructure other than the configuration. 

Thanks!

**Comment by bmcelvee** [comment]
Preview build (should update shortly): https://osdocs-1192-azure-gov-reg--ocpdocs.netlify.app/openshift-enterprise/latest/registry/configuring_registry_storage/configuring-registry-storage-azure-user-infrastructure.html

**Comment by bmcelvee** [comment]
@xiuwang would you mind reviewing this PR, please? Thank you!

File: modules/registry-configuring-storage-azure-gov-cloud.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+----
++
+.Example configuration
+[source,yaml]
+----
+Storage:
+Azure:
+Account Name: imageregistryadcd
+Cloud Name: AzureUSGovernmentCloud
+Container: <container-name>
```
Comment (dmage) [inline modules/registry-configuring-storage-azure-gov-cloud.adoc:37]:
```yaml
storage:
  azure:
    accountName: <storage-account-name>
    container: <container-name>
    cloudName: AzureUSGovernmentCloud
```

File: modules/registry-configuring-storage-azure-gov-cloud.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+[id="registry-configuring-storage-azure-gov-cloud_{context}"]
+= Configuring registry storage for Azure Government Cloud
+
+During installation, your cloud credentials are sufficient to create Azure Blob
+Storage, and the Registry Operator automatically configures storage.
+
+.Prerequisites
+
+* A cluster on Azure with Government Cloud infrastructure.
```
Comment (dmage) [inline modules/registry-configuring-storage-azure-gov-cloud.adoc:13]: `A cluster on Azure with user-provisioned infrastructure in a government region`?

**Comment by dmage** [comment]
@bmcelvee basically there is no difference, it's just a different region (environment?). Azure IPI and Azure Government IPI as very similar: the storage is automatically provisioned and no extra steps are needed. But there is a possibility to use a manually created container (mainly in case of UPI, I guess?). To use a government cloud container, the customer should add an extra parameter to the configuration: `cloudName: AzureUSGovernmentCloud`.

File: modules/registry-configuring-storage-azure-gov-cloud.adoc (hunk 1 above)
Comment (dmage) [reply modules/registry-configuring-storage-azure-gov-cloud.adoc:37]: btw, there are few more options: https://github.com/openshift/api/blob/43bf8c1cc69ff4f5468d22b5ffb241666aee77a8/config/v1/types_infrastructure.go#L300

* AzurePublicCloud (the default value if not set)
* AzureUSGovernmentCloud
* AzureChinaCloud
* AzureGermanCloud

**Comment by bmcelvee** [comment]
Thank you, @dmage!

File: _topic_map.yml [hunk 3]
```diff
@@ -804,7 +804,7 @@ Topics:
     File: configuring-registry-storage-gcp-user-infrastructure
 #  - Name: Configuring the registry for OpenStack user-provisioned infrastructure
 #    File: configuring-registry-storage-openstack-user-infrastructure
-  - Name: Configuring the registry for Azure user-provisioned infrastructure
+  - Name: Configuring the registry for Azure
```
Comment (codyhoag) [inline _topic_map.yml:807]: I noticed the gov section also has a prereq for a cluster with UPI. Should the `user-provisioned infrastructure` be added back here?

File: modules/registry-configuring-storage-azure-gov-cloud.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
+// Module included in the following assemblies:
+//
+//* registry/configuring_registry_storage-azure.adoc
+
+[id="registry-configuring-storage-azure-gov-cloud_{context}"]
+= Configuring registry storage for Azure Government Cloud
```
Comment (codyhoag) [inline modules/registry-configuring-storage-azure-gov-cloud.adoc:6]: Totally a judgment call, but unlike AWS GovCloud, I've usually only seen Azure's gov region product called [Azure Government](https://azure.microsoft.com/en-us/global-infrastructure/government/get-started/) (or MAG). With that being said, the *Cloud* part could be dropped.

File: modules/registry-configuring-storage-azure-gov-cloud.adoc [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+During installation, your cloud credentials are sufficient to create Azure Blob
+Storage, and the Registry Operator automatically configures storage.
+
+.Prerequisites
+
+* A cluster on Azure with user-provisioned infrastructure in a government region.
+* To configure registry storage for Azure, provide Registry Operator
+cloud credentials.
+* For Azure storage the secret is expected to contain one key:
```
Comment (codyhoag) [inline modules/registry-configuring-storage-azure-gov-cloud.adoc:16]: s/For Azure storage the/For Azure storage, the

File: registry/configuring_registry_storage/configuring-registry-storage-azure-user-infrastructure.adoc [hunk 6]
```diff
@@ -1,5 +1,5 @@
 [id="configuring-registry-storage-azure-user-infrastructure"]
-= Configuring the registry for Azure user-provisioned infrastructure
+= Configuring the registry for Azure
```
Comment (codyhoag) [inline registry/configuring_registry_storage/configuring-registry-storage-azure-user-infrastructure.adoc:2]: Same comment as I made for topic map

File: modules/registry-configuring-storage-azure-gov-cloud.adoc [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+.Example configuration
+[source,yaml]
+----
+storage:
+  azure:
+    accountName: <storage-account-name>
+    container: <container-name>
+    cloudName: AzureUSGovernmentCloud <1>
+----
+<1> `cloudName`: Defaults to `AzurePublicCloud`. You can also set `cloudname` to `AzureUSGovernmentCloud`, `AzureChinaCloud`, or `AzureGermanCloud`.
```
Comment (codyhoag) [inline modules/registry-configuring-storage-azure-gov-cloud.adoc:39]: What do you think about stripping the \`cloudName\`: part from the front? Perhaps:

> <1> Defaults to \`AzurePublicCloud\`. You can also set \`cloudname\` to \`AzureUSGovernmentCloud\`, \`AzureChinaCloud\`, or \`AzureGermanCloud\`.

**Review by codyhoag — COMMENTED** [review]
A few nits; otherwise, LGTM!

File: modules/registry-configuring-storage-azure-gov-cloud.adoc [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+.Example configuration
+[source,yaml]
+----
+storage:
+  azure:
+    accountName: <storage-account-name>
+    container: <container-name>
+    cloudName: AzureUSGovernmentCloud <1>
+----
+<1> Defaults to `AzurePublicCloud`. You can also set `cloudname` to `AzureUSGovernmentCloud`, `AzureChinaCloud`, or `AzureGermanCloud`.
```
Comment (xiuwang) [inline modules/registry-configuring-storage-azure-gov-cloud.adoc:39]: Better to add the explain for cloudName, as https://github.com/openshift/api/pull/650/files#diff-3ff94e87467e8743ff79a40fa85f123d3205a560f79b4dea1919e76a4d55958dR230-R233.

Changing the cloudName need sufficient credentials. I'm afraid this description is not rigorous enough.
How about "You can also set `cloudname` to `AzureUSGovernmentCloud`, `AzureChinaCloud`, or `AzureGermanCloud` with sufficient credentials" ?

**Pull request merged**